tests/test_edge_cases.py

```diff
@@ -880,3 +881,52 @@ class Spam(BaseModel):
     assert Spam(c=Foo(a='123')).dict() == {'c': {'a': 123}}
     with pytest.raises(ValidationError):
         Spam(c=Bar(b='123'))
+
+
+def test_orm_mode():
+    class PetCls:
+        def __init__(self, *, name: str, species: str):
+            self.name = name
+            self.species = species
+
+    class PersonCls:
+        def __init__(self, *, name: str, age: float = None, pets: List[PetCls]):
+            self.name = name
+            self.age = age
+            self.pets = pets
+
+    class Pet(BaseModel):
+        name: str
+        species: str
+
+        class Config:
+            orm_mode = True
+
+    class Person(BaseModel):
+        name: str
+        age: float = None
+        pets: List[Pet]
+
+        class Config:
+            orm_mode = True
+
+    bones = PetCls(name='Bones', species='dog')
+    orion = PetCls(name='Orion', species='cat')
+    anna = PersonCls(name='Anna', age=20, pets=[bones, orion])
+
+    anna_model = Person.from_orm(anna)
+
+    assert anna_model.dict() == {
+        'name': 'Anna',
+        'pets': [{'name': 'Bones', 'species': 'dog'}, {'name': 'Orion', 'species': 'cat'}],
+        'age': 20.0,
+    }
+
+
+def test_not_orm_mode():
+    class Pet(BaseModel):
+        name: str
+        species: str
+
+    with pytest.raises(ConfigError):
+        Pet.from_orm(None)
```

Comment:
Could you add this test case? (it's already passing).

It would ensure that we don't break dynamic properties and other stuff ORMs do:

```Python
def test_object_with_getattr():
    class FooGetAttr:
        def __getattr__(self, key: str):
            if key == 'foo':
                return 'Foo'
            else:
                raise AttributeError

    class Model(BaseModel):
        foo: str
        bar: int = 1

        class Config:
            orm_mode = True

    class ModelInvalid(BaseModel):
        foo: str
        bar: int

        class Config:
            orm_mode = True

    foo = FooGetAttr()
    model = Model.from_orm(foo)
    assert model.foo == 'Foo'
    assert model.bar == 1
    assert model.dict(skip_defaults=True) == {'foo': 'Foo'}
    with pytest.raises(ValidationError):
        ModelInvalid.from_orm(foo)
```